### PR TITLE
Feature/rework simulator with shared model

### PR DESCRIPTION
**Pull request by Alexey-N-Chernyshov**
### Description of the Change
Rework simulator with shared_model.

### Benefits
Less old model dependencies.

### Possible Drawbacks 
No

### Usage Examples or Tests *[optional]*
no

### Alternate Designs *[optional]*
no

### CODE REVIEW

File: irohad/simulator/impl/simulator.cpp [hunk 1]
```diff
@@ -92,21 +94,36 @@ namespace iroha {
           });
     }
 
-    void Simulator::process_verified_proposal(model::Proposal proposal) {
+    void Simulator::process_verified_proposal(
+        const shared_model::interface::Proposal &proposal) {
       log_->info("process verified proposal");
+
       model::Block new_block;
```
Comment (Alexey-N-Chernyshov) [inline irohad/simulator/impl/simulator.cpp]: Block will be replaced with crypto_provider relocation. Now it depends on signature and IR-968 will resolve this dependency.

File: irohad/simulator/impl/simulator.cpp [hunk 2]
```diff
@@ -33,15 +37,18 @@ namespace iroha {
           block_queries_(std::move(blockQuery)),
           crypto_provider_(std::move(crypto_provider)) {
       log_ = logger::log("Simulator");
-      ordering_gate->on_proposal().subscribe(proposal_subscription_,
-                                             [this](model::Proposal proposal) {
-                                               this->process_proposal(proposal);
-                                             });
+      ordering_gate->on_proposal().subscribe(
+          proposal_subscription_, [this](model::Proposal old_proposal) {
+            auto proposal = shared_model::proto::Proposal(
```
Comment (kamilsa) [inline irohad/simulator/impl/simulator.cpp]: why constructor here?

File: irohad/consensus/yac/impl/yac_gate_impl.cpp [hunk 3]
```diff
@@ -42,8 +42,9 @@ namespace iroha {
             block_loader_(std::move(block_loader)),
             delay_(delay) {
         log_ = logger::log("YacGate");
-        block_creator_->on_block().subscribe(
-            [this](auto block) { this->vote(block); });
+        block_creator_->on_block().subscribe([this](auto block) {
```
Comment (Warchant) [inline irohad/consensus/yac/impl/yac_gate_impl.cpp]: one block copy

File: irohad/consensus/yac/impl/yac_gate_impl.cpp [hunk 4]
```diff
@@ -42,8 +42,9 @@ namespace iroha {
...
+          this->vote(*std::unique_ptr<iroha::model::Block>(block->makeOldModel()));
```
Comment (Warchant) [inline irohad/consensus/yac/impl/yac_gate_impl.cpp]: this->vote is another block copy

File: irohad/consensus/yac/impl/yac_gate_impl.cpp (hunk 3 above)
Comment (kamilsa) [reply irohad/consensus/yac/impl/yac_gate_impl.cpp]: block is `shared_ptr` passing by value is acceptable and more thread-safe

**Pull request merged**